### PR TITLE
Add roll_dice function

### LINKED ISSUE #216: fn roll_dice
Extract things like `thread_rng().gen_range(0, 6)` to a special function.

### PR DESCRIPTION
**Pull request by PatriotRossii**
Closes #216 (fn roll_dice)

### CODE REVIEW

File: src/core/utils.rs [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+    distributions::uniform::{SampleBorrow, SampleUniform},
+    seq::SliceRandom,
+    Rng,
+};
 
 pub fn zrng() -> impl rand::Rng {
     QuadRand
 }
 
+pub fn roll_dice<T: SampleUniform, B1, B2>(low: B1, high: B2) -> T
```
Comment (ozkriff) [inline src/core/utils.rs:14]: Why different B1 and B2 types for each argument are required?

Comment (PatriotRossii) [reply src/core/utils.rs:14]: > Why different B1 and B2 types for each argument are required?

Idk, but it's the signature of rand::Rng::gen_range(). I decided to use it for compatibility with its use cases.

Comment (ozkriff) [reply src/core/utils.rs]: Ok, I'll simplify the signature then :)

**Comment by ozkriff** [comment]
Merged, thanks for the PR! Sorry for the delay, was busy with dayjob and the newsletter.

**Pull request merged**